### ISSUE TITLE
Missing features in new version

### ISSUE THREAD

**Issue by X-Looper** [Hunter5000/twitch-fox#1]
First, let me just say that I love this add-on. I've been using it for quite some time now and it has really improved my stream viewing on Twitch significantly ever since and I'm deeply grateful to you for creating this amazing add-on, many thanks Hunter5000!

Now, let me go over some of my issues with the latest version. I'll try and keep this as short and brief as I can.

**• Not being able to follow Games / Channels without logging onto Twitch**

I know others have already given you feedback about this and I noticed that you made a quick update again that allows you to follow Channels.

However, there doesn't seem to be an option to follow specific Games anymore and this was one of my favorite features, due to me being an old-school gamer. I used Twitch Fox to add a whole bunch of old and obscure games to my follow list, just in case someone would stream those games one day and would love to see that feature return in this new version.

**• Old Twitch Fox Follow List incompatible with new version**

When I try and use the "Import follows from file" and select my old Follow List from the old version of Twitch Fox, nothing happens, as it would seem that the list is no longer compatible with the new version.

**• No longer seems to support Streamlink / Livestreamer**

This is a big one. I stopped watching Livestreams through the browser many years ago thanks to Livestreamer (now using Streamlink) by watching all my livestreams through MPC-HC instead for a much smoother experience and I have no desire of ever going back to browser-based viewing of livestreams.

In this new version there doesn't seem to be any way to launch the livestreams with Streamlink anymore (which you could with the old version) and I really do hope that you will add this feature to the new version as well, as for me personally, it's a MUST have.
___________________________________________________________________________________________________________

Lastly, I have a few smaller requests for features I would love to see one day, nothing important, just some options / features I wish Twitch Fox had.

**+ Different ways to sort the list of Streamers**

I would love to have the option to sort the list of Streamers by number of viewers, stream duration, etc.

For example, I select a certain game and it has over 500 different people streaming it and I just want to watch someone who has 10 or lower viewers. Now, instead of having to scroll all the way to the bottom of the entire list, I would be able to sort the list so the top starts with 0 viewers and the number of viewers go up as you scroll down the list instead, as well as being able to sort the streams based on how long they've been streaming.

**+ Improved Search / Filtering of Streamers**

I love the fact that you can filter out the list of streamers by typing in a word in the search field above, as it really helps you narrow down the list of streamers when looking for something specific in the list.

For example, I head over to Street Fighter V and there are over 500 people streaming the game and I would like to watch someone who's playing Chun-li, I would type Chun-li in the search field above and every livestream with the word Chun-li in their title would be displayed in the list.

However, the way things are now, you would first have to scroll down ALL the way to the bottom of the entire list of 500 streamers BEFORE you can type in Chun-li in the search field above, as the search only filters the livestreams that are currently being displayed in your list.

Now, what I would love, is if you could make it so that Twitch Fox would search and filter through the entire list of ALL streamers in one go and without having to first scroll down the entire list to the very bottom (which can take a while if it's a popular game with many streamers).
________________________________________________________________________________________________________

That's all I can think of for now and once again, thank you so much for this great add-on, Hunter5000!

**Comment by Hunter5000** [comment]
Thank you for putting your suggestions together so nicely! It makes it much easier to focus on what to do. I will try to briefly address all of them.

**Followed Games**
I will try to address this somehow. The Twitch API endpoint is unofficial and undocumented, but I don't even have to use it if I make a workaround.

**Old Twitch Fox Followed List Compatibility**
This may be kind of difficult and may have some room of error, but I will try to get this working.

**Streamlink/Livestreamer**
I miss this feature the most personally. The only reason it's gone is because I don't know how to implement it with WebExtensions yet. I will try my best to figure this out, and it will probably be using Streamlink rather than Livestreamer.

**Sorting the list**
This shouldn't be too hard to add back. I'm not even really sure why I didn't add this feature before the 4.0.0 release, I guess I was just too excited.

**Improved search**
I'll see if this is possible given the Twitch API endpoint specifications. It should be in some cases, and where it is I will make an effort to implement it.

Now, I'm not sure when I will actually be adding these features in, but hopefully not too long from now. Some of them are much easier to add in than others, so I'll try to get some of the simpler ones out of the way when I find the time.

**Comment by X-Looper** [comment]
I really wish I could do more than just request, but sadly I don't know anything about programming I'm afraid and as such, I don't even know what's possible to implement and what's not. So for me, watching you people create these amazing add-ons is like watching wizards working their magic, all I can really do is sit back and gaze in awe.

Also, please don't feel like you need to address any and all of these things at once. I just wanted to give some feedback on the new version along with a couple of requests. I know you've been working hard trying to get this new version out for some time now and I can imagine trying to keep add-ons updated and working alongside Firefox's own updates can be a lot of work.

I'm just happy to see that you haven't abandoned this great add-on, as I don't even want to imagine having to go back to browsing Twitch the old-fashioned way ever again. Yeah, you could say I have a bit of a Twitch Fox addiction, hehe.

**Comment by Chamlotte** [comment]
I, too, love this addon, and I'm very, very happy it got updated for 57. I figure a dark theme is already in the works, but I do have a layout question.

Looking at the list of live followed channels, is there a way to have a setup similar to the old layout with the large streamer avatar? That's usually the first thing I would look at to identify a streamer, and their names are hard to make out and kind of a bit too out of the way in comparison to the old theme. If it's not possible or too much work I can get used to it I think, especially after a dark theme happens.

**Comment by Hunter5000** [comment]
Indeed, a dark mode is in the works, and assuming it's as easy as I predict, it should be in tomorrow. As for the layout request, it should be possible I think. I'll probably just parse a stream as I would a channel and use that layout. I'm not sure where I would put the "popout" button since there is less space but I will figure out something.

**Comment by Chamlotte** [comment]
TBH the current layout works fine, just maybe putting the avatar next to the stream preview would make them more distinguishable than a small line of text under them.

**Comment by Hunter5000** [comment]
Okay, version 4.2.1 has dark mode and small (optional) channel logos.
